### PR TITLE
Initial commit for Help command (doesn't compile)

### PR DESCRIPTION
**Pull request by artem-kim**
Hi team, I am pulling my hair out a bit with trying to make the code run. I added a "HelpCommand.java" to the commands folder (with some dummy method there). Could somebody have a look at the errors and help me troubleshoot it? 

[ERROR] /Users/wwhd/Documents/SE2/newbank3/newbank/src/main/java/newbank/server/NewBankClientHandler.java:[52,26] incompatible types: invalid constructor reference
[ERROR]     constructor HelpCommand in class newbank.server.commands.HelpCommand cannot be applied to given types
[ERROR]       required: no arguments
[ERROR]       found:    newbank.server.NewBank,java.lang.String[],newbank.server.CustomerID
[ERROR]       reason: actual and formal argument lists differ in length

### CODE REVIEW

**Comment by michowski** [comment]
Hi Artem, the IDE/editor should show you:

```
The type HelpCommand does not define HelpCommand(NewBank, String[], CustomerID) that is applicable hereJava(603979903)
```

in `src/main/java/newbank/server/NewBankClientHandler.java:52`

**Comment by michowski** [comment]
@artem-kim I've added and pushed the missing constructor. Could you please do `git pull origin 42-ui-update-v2`?

**Comment by artem-kim** [comment]
@michowski @juliano60 @ZacTredger @claudist good morning team, I finished the HelpCommand, added welcome instructions and updated the protocol with a new help command. Please review.

File: src/main/java/newbank/server/NewBankClientHandler.java
Comment (juliano60) [inline src/main/java/newbank/server/NewBankClientHandler.java]: This needs to move to the client side.

Regards,
Jules

File: src/main/java/newbank/client/ExampleClient.java
Comment (juliano60) [inline src/main/java/newbank/client/ExampleClient.java]: Please catch up with main as @claudist has already checked that in.

Regards,
Jules

File: src/main/java/newbank/server/commands/HelpCommand.java
Comment (juliano60) [inline src/main/java/newbank/server/commands/HelpCommand.java:11]: Rather than displaying the syntax name, which they can get by issuing "COMMAND HELP", we could display,
"COMMAND  a brief explanation of the command".
Try "git help" on the command line and use the output for inspiration.

Regards,
Jules

File: docs/protocol.md
Comment (juliano60) [inline docs/protocol.md]: The example does not make sense for this command. The output is too long and not an example as it doesn't change.
Note that it says that you need to be logged in in the output which is incorrect.
The command also should return "SUCCESS or FAIL" probably.

Kindest Regards,
Jules

**Review by juliano60 — CHANGES_REQUESTED** [review]
Good attempt @artem-kim , a few comments to address.

Kindest Regards,
Jules

File: src/main/java/newbank/client/ExampleClient.java
Comment (juliano60) [reply src/main/java/newbank/client/ExampleClient.java]: Resolved.

File: src/main/java/newbank/server/NewBankClientHandler.java
Comment (juliano60) [reply src/main/java/newbank/server/NewBankClientHandler.java]: Fixed.

Regards,
Jules

**Comment by juliano60** [comment]
@artem-kim I resolved some of the issues (caught your branch up with "main" and moved some stuffs to the client side).
Still some outstanding issues for you to look at. Just catch up with origin/42-ui-update-v2 to download the changes.

Kindest Regards,
Jules

**Comment by juliano60** [comment]
@michowski  @ZacTredger Could you please review?
= tidied up the HELP command
= added fixes for SHOWCCOUNTS

Actually forgot something, hold off the review for now...

Regards,
Jules

**Comment by juliano60** [comment]
image

File: docs/protocol.md
Comment (ZacTredger) [inline docs/protocol.md]: You close the `code` tag twice. Better to make sure though, eh?

File: docs/protocol.md
Comment (ZacTredger) [inline docs/protocol.md]: Should this be `SUCCESS` instead of `info`?

File: src/main/java/newbank/utils/Config.java
Comment (ZacTredger) [inline src/main/java/newbank/utils/Config.java:4]: This was a great idea :+1:

File: src/test/java/newbank/TestApp.java
Comment (ZacTredger) [inline src/test/java/newbank/TestApp.java]: I feel like this pattern is repeatedly crying out "extract method!" No?

**Review by ZacTredger — DISMISSED** [review]
Added some teeny tiny suggestions. Only the `info` vs `SUCCESS` thing in `protocol.md` is actually observable to users.
I'm approving, but if you want, I can just add another commit to fix these little things? Let me know.
Great work guys
Z

**Pull request merged**
